quickwit-doc-mapping/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -0,0 +1,605 @@
+            }
+            FieldMappingType::Bytes(options) => {
+                stored = options.is_stored();
+                indexed = options.is_indexed();
+                fast = options.is_fast();
+            }
+            _ => (),
+        }
+
+        FieldMappingEntryForSerialization {
+            name: value.name,
+            type_with_cardinality,
+            fast,
+            indexed,
+            record,
+            stored,
+            tokenizer,
+            field_mappings,
+        }
+    }
+}
+
+impl FieldMappingEntryForSerialization {
+    fn is_array(&self) -> bool {
+        self.type_with_cardinality.starts_with("array<")
+            && self.type_with_cardinality.ends_with('>')
+    }
+
+    fn field_type(&self) -> &str {
+        if self.is_array() {
+            &self.type_with_cardinality[6..self.type_with_cardinality.len() - 1]
+        } else {
+            &self.type_with_cardinality
+        }
+    }
+
+    fn new_text(&self) -> anyhow::Result<FieldMappingType> {
+        if self.is_array() {
+            bail!(format!(
+                "Error when parsing field `{}`: array of text is not supported",
+                self.name
+            ))
+        }
+        let mut indexing_options = TextFieldIndexing::default();
+        if let Some(index_option) = self.record {
+            indexing_options = indexing_options.set_index_option(index_option);
+        }
+        if let Some(tokenizer) = &self.tokenizer {
+            indexing_options = indexing_options.set_tokenizer(tokenizer);
+        }
+        let mut options = TextOptions::default().set_indexing_options(indexing_options);
+        if self.stored {
+            options = options.set_stored();
+        }
+        Ok(FieldMappingType::Text(options))
+    }
+
+    fn new_i64(&self) -> anyhow::Result<FieldMappingType> {
+        let options = self.int_options()?;
+        Ok(FieldMappingType::I64(options))
+    }
+
+    fn new_f64(&self) -> anyhow::Result<FieldMappingType> {
+        let options = self.int_options()?;
+        Ok(FieldMappingType::F64(options))
+    }
+
+    fn new_date(&self) -> anyhow::Result<FieldMappingType> {
+        let options = self.int_options()?;
+        Ok(FieldMappingType::Date(options))
+    }
+
+    fn new_bytes(&self) -> anyhow::Result<FieldMappingType> {
+        self.check_no_text_options()?;
+        let mut options = BytesOptions::default();
+        if self.stored {
+            options = options.set_stored();
+        }
+        if self.indexed {
+            options = options.set_indexed();
+        }
+        Ok(FieldMappingType::Bytes(options))
+    }
+
+    fn new_object(&self) -> anyhow::Result<FieldMappingType> {
+        if self.record.is_some() || self.tokenizer.is_some() {
+            bail!(format!(
+                "Error when parsing field `{}`: `field_mappings` is the only valid parameter.",
+                self.name
+            ))
+        }
+        if self.is_array() {
+            bail!(format!(
```

Comment:
nitpick bail includes formatting.
```
bail!("Error when parsing field `{}`: array of object is not supported", self.name)
```

should work

Reply:
ah yes I always forget it